packages/provider-keeper/tests/ui.spec.ts

```diff
@@ -0,0 +1,396 @@
+        tabWavesKeeper = await driver.getWindowHandle();
+
+        await driver.switchTo().newWindow('tab');
+        await driver.get(`http://${ui.host}:${ui.port}`);
+        await driver.wait(until.elementLocated(By.id('sign-in')), timeout);
+        tabUI = await driver.getWindowHandle();
+        await driver.sleep(2000);
+    });
+
+    after(async function () {
+        this.timeout(60 * s);
+        driver && (await driver.quit());
+        cSelenium && (await cSelenium.stop());
+        cUI && (await cUI.stop());
+    });
+
+    it('auth tx', async () => {
+        await driver.switchTo().window(tabUI);
+
+        const loginBtn = await driver.wait(until.elementLocated(By.css('#sign-in:not(.disabled)')), timeout);
+        await loginBtn.click();
+
+        await driver.switchTo().window(tabWavesKeeper);
+        // site permission request
+        await driver.wait(until.elementLocated(By.xpath("//div[contains(@class, '-originAuthTx')]")));
+        let acceptBtn = await driver.findElement(By.css('.app button[type=submit]'));
+        await acceptBtn.click();
+        // site auth request
+        await driver.wait(until.elementLocated(By.xpath("//div[contains(@class, '-authTx')]")));
+        acceptBtn = await driver.findElement(By.css('.app button[type=submit]'));
+        await acceptBtn.click();
+        // close window
+        const closeBtn = await driver.wait(until.elementLocated(By.xpath("//button[contains(@class, '-closeBtn')]")));
+        await closeBtn.click();
+
+        await driver.switchTo().window(tabUI);
+        const userData = await driver.executeScript(() => {
+            return (window as any).getOutput();
+        });
+        expect(userData.signature).to.exist;
+    });
+
+    const isValidSignedTx = async (formSelector, tx) => {
+        await driver.switchTo().window(tabUI);
+        await driver.executeScript((tx) => {
+            (window as any).setInput(tx);
+        }, tx);
+        const sendTxBtn = await driver.wait(until.elementLocated(By.css('#send-tx:not(.disabled)')), timeout);
+        await sendTxBtn.click();
+
+        // tx request
+        await driver.switchTo().window(tabWavesKeeper);
+        await driver.wait(until.elementLocated(formSelector));
+        const acceptBtn = await driver.findElement(By.css('.app button[type=submit]'));
+        await acceptBtn.click();
+        // close window
+        const closeBtn = await driver.wait(until.elementLocated(By.xpath("//button[contains(@class, '-closeBtn')]")));
+        await closeBtn.click();
+
+        await driver.switchTo().window(tabUI);
+        await driver.wait(until.elementLocated(By.css('#send-tx.disabled')), timeout);
+        const signed = await driver.executeScript(() => {
+            return (window as any).getOutput();
+        });
+        expect(signed.length).to.be.equal(1);
+        const signedTx = signed[0];
+        expect(signedTx.id).to.exist;
+        expect(signedTx.type).to.exist;
```

Comment:
Can be compared with the fields of the input `tx`? `expect(signedTx.type).to.be.equal(tx.type)` and so on.

And also validate resulting signature.